native-sql-engine/core/src/main/scala/com/intel/oap/expression/ColumnarExpressionConverter.scala

```diff
@@ -245,12 +246,31 @@ object ColumnarExpressionConverter extends Logging {
           expr)
       case u: UnaryExpression =>
         logInfo(s"${expr.getClass} ${expr} is supported, no_cal is $check_if_no_calculation.")
-        ColumnarUnaryOperator.create(
-          replaceWithColumnarExpression(
-            u.child,
-            attributeSeq,
-            convertBoundRefToAttrRef = convertBoundRefToAttrRef),
-          expr)
+        if (!u.isInstanceOf[CheckOverflow] || !u.child.isInstanceOf[Divide]) {
+          ColumnarUnaryOperator.create(
+            replaceWithColumnarExpression(
+              u.child,
+              attributeSeq,
+              convertBoundRefToAttrRef = convertBoundRefToAttrRef),
+            expr)
+        } else {
+          // CheckOverflow[Divide]: pass resType to Divide to avoid precision loss
```

Comment:
does this mean divide must have a checkoverflow wrapper?

Reply:
look like yes for decimal divide. Below is spark code from spark/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/DecimalPrecision.scala
`      CheckOverflow(Divide(promotePrecision(e1, widerType), promotePrecision(e2, widerType)),
        resultType, nullOnOverflow)`